### DISCUSSION TITLE
How to configure targets/endpoints

### DISCUSSION THREAD

**Discussion by BlueSeaOcean** [cloudprober/cloudprober#630] [Q&A]
I would like to configure full url like "www.helloworld.com:1234/v1/test". Can you advice what the config would look like?
I tried to configure as host_names under target, but got an error "targets.New(): error creating targets from host_names: invalid host (www.helloworld.com:1234/v1/test), contains '/' ".
```
probe {
 name: "hello-world"
 type: ENTENSION
 targets {
  host_names: "www.helloworld.com:1234/v1/test"
 }
 [myprobe] {
   op: "set"
   key: "testkey"
   value: "testval"
 }
}
```

**Comment by manugarg** [comment]
@BlueSeaOcean I documented it here:
https://cloudprober.org/docs/how-to/targets/#static-targets

Since you're using probe extension, you'll need additional information. Configuring targets like this, adds following labels to the targets:
* `__cp_scheme__`
* `__cp_host__`
* `__cp_path__`

Port is already part of the target.

See this code to get an idea of how endpoints are parsed:
https://github.com/cloudprober/cloudprober/blob/b6174266707a875d2bd6cf0f8f5fa729fb4ecdcf/targets/endpoint/endpoint.go#L154
and 
https://github.com/cloudprober/cloudprober/blob/b6174266707a875d2bd6cf0f8f5fa729fb4ecdcf/probes/http/request.go#L70
for how these labels are used by the stock HTTP probe.

**Reply by BlueSeaOcean** [reply]
I looked into the how-to document and tried followings. Can you help point out how it should look like exactly?

unknown field name `"endpoints"` in proto.TargetsDef
```
probe {
 name: "hello-world"
 type: ENTENSION
 targets {
  endpoints {
   url: "www.helloworld.com:1234/v1/test"
  }
 }
 [myprobe] {
   op: "set"
   key: "testkey"
   value: "testval"
 }
}
```
and
unknown field name `"__cp_path__"` in proto.TargetsDef
```
probe {
 name: "hello-world"
 type: ENTENSION
 targets {
  hostnames: "www.helloworld.com:1234"
  __cp_path__: "/v1/test"
 }
 [myprobe] {
   op: "set"
   key: "testkey"
   value: "testval"
 }
}
```

**Reply by manugarg** [reply]
`endpoints` work only with the latest version: v0.13.1. 

You don't need to specify `__cp_path__`. It will be computed from the URL automatically.

Also, endpoints require a name.
